### DISCUSSION TITLE
how to run the pipeline on the cpu ?

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1291] [Development]
I'm trying to run pyannote and getting error: "CUDA out of memory".

so,  I want to run my code on CPU rather than GPU.

```
from pyannote.audio import Pipeline

pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization",use_auth_token="token...")

diarization = pipeline("example.wav")
```

How can I run the speaker-diarization pipeline on the CPU given the above configuration?

**Comment by MustafaCQN** [comment]
Hi @amitli1 

please check this thread
https://github.com/pyannote/pyannote-audio/discussions/1095

you can specify the device type as `pipeline.to(device)`

**Reply by amitli1** [reply]
It doesn't work:
```
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization",use_auth_token=MY_TOKEN)
pipeline.to("cpu")
```

Error:
`AttributeError: 'SpeakerDiarization' object has no attribute 'to'`

I'm using the following versions:
```
pyannote.audio==2.1.1
pyannote.core==4.5
pyannote.database==4.1.3
pyannote.metrics==3.2.1
pyannote.pipeline==2.3
```

**Reply by MustafaCQN** [reply]
Can you share your 
`print(torch.__version__)` 
and
`print(torch.cuda.is_available()`
sorry for earlier answer, it will work on offlline models. Seems you are using an online model

**Reply by amitli1** [reply]
Results fot the versions & cuda:
```
1.13.1+cu117
True
```

How can I use pyannote pipeline offlline models ?

**Reply by MustafaCQN** [reply]
> 
> How can I use pyannote pipeline offlline models ?

You can use the Segmentation model. Downloadable from HuggingFace.

> Results fot the versions & cuda:
> 
> ```
> 1.13.1+cu117
> True
> ```
You can do a tweak in this operation. You can delete your torch and re-install the cpu only version in order to run it from cpu

**Reply by amitli1** [reply]
Segmentation model is just one model of the `pyannote/speaker-diarization` (which contains Segmentation + Embedding + Clusterring) .
So download just it from `HuggingFace` is not sufficient, am I right ?
